### DISCUSSION TITLE
About the ionospheric VTEC output value

### DISCUSSION THREAD

**Discussion by gnssxx** [groops-devs/groops#151] [Q&A]
The ionospheric VTEC output value included in the parameter setting section of the 030groovsGnssProcessing.xml file in scenarioGnssNetwork. Is the output ionospheric VTEC value the total vertical electron content of ionospheric delay? Or remove the remaining vertical total electron content of the model value? I checked your paper, but I still don't understand it. I hope you can explain it further. Thank you very much!

**Comment by tmayerguerr** [comment]
I am unsure if I have understood the question correctly.
Currently, no apriori model is introduced for the ionosphere. So estimated VTEC values are always the total vertical total electron content. To do this, the (slant) STEC values are mapped into the vertical using the mapping function mentioned in [gnssParametrization:VTEC](https://groops-devs.github.io/groops/html/gnssParametrizationType.html#ionosphereVTEC). Residual local and short-term scintillations should be estimated additionally with contrained [gnssParametrization:STEC](https://groops-devs.github.io/groops/html/gnssParametrizationType.html#ionosphereSTEC).

regards
Torsten

**Reply by realrxguo** [reply]
I also checked your paper [ref](https://link.springer.com/article/10.1007/s00190-018-1223-2) but I still don't understand it. Can you explain the specific strategy for dealing with the ionosphere? especially in LEO orbit determination. Thanks!

**Comment by tmayerguerr** [comment]
It is still unclear to me what your goal is. Do you just want to determine LEO orbits and eliminate the influence of the ionosphere? Or are you interested in the estimated TEC values itself? In the first case, you simply have to estimate a [gnssParametrization:STEC](https://groops-devs.github.io/groops/html/gnssParametrizationType.html#ionosphereSTEC) parameter for each observation and no other TEC related parametrizations have to be setup. This corresponds to the ionosphere-free linear combination in other software packages.

regards
Torsten

**Reply by realrxguo** [reply]
Dear  Torsten
     I would like to understand the ionospheric processing strategy in the examples you provided for LEO kinematics   and reduced dynamic orbit determination [https://ftp.tugraz.at/outgoing/ITSG/groops/scenario/](url). The STEC parameters are eliminated before the cumulative equation, similar to a linear combination without ionosphere，can I understand it in the above way ?  But when I was debugging the program, I found that after pre eliminating the ionospheric parameter STEC, why do the corresponding STEC parameters coefficient still exist in the original observation equation?  It's just that it's numerically small.  And it is multiplied by the corresponding projection function in the subsequent accumulate Normals to become the coefficient matrix of VTEC. I hope you can explain it further. Thank you very much!

**Comment by tmayerguerr** [comment]
Sorry for the late answer, it was a very busy time.

The complete observation equations are setup in two steps in GROOPS. In a first step basic GnssObservationEquations are computed and in second step these are passed to GnssParametrizations. The STEC parameter in GnssObservationEquation is used twice: On the one hand for estimating the STEC itself (directly eliminated from the equations) and for the other to multiply it with the VTEC mapping. Estimating unconstrained STEC and VTEC parameters together are meaningless as you already mentioned after elimination the partials are numerically zero for the unconstrained STEC. It is on user side to select a meaningful combination of parametrizations. So, you should add a constraint (e.g. 5/sin(E)) to STEC when you additionally setup VTEC parameters.

regards
Torsten
